x-pack/plugins/monitoring/public/alerts/panel.tsx

```diff
@@ -0,0 +1,196 @@
+import { replaceTokens } from './lib/replace_tokens';
+import { AlertsContextProvider } from '../../../triggers_actions_ui/public';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { AlertEdit } from '../../../triggers_actions_ui/public';
+import { isInSetupMode } from '../lib/setup_mode';
+import { BASE_ALERT_API_PATH } from '../../../alerts/common';
+
+interface Props {
+  alert: CommonAlertStatus;
+}
+export const AlertPanel: React.FC<Props> = (props: Props) => {
+  const {
+    alert: { states, alert },
+  } = props;
+  const [showFlyout, setShowFlyout] = React.useState(false);
+  const [isEnabled, setIsEnabled] = React.useState(alert.rawAlert.enabled);
+  const [isMuted, setIsMuted] = React.useState(alert.rawAlert.muteAll);
+  const inSetupMode = isInSetupMode();
+
+  if (!alert.rawAlert) {
+    return null;
+  }
+
+  async function disableAlert() {
+    try {
+      await Legacy.shims.http.post(`${BASE_ALERT_API_PATH}/alert/${alert.rawAlert.id}/_disable`);
+    } catch (err) {
+      Legacy.shims.toastNotifications.addDanger({
+        title: i18n.translate('xpack.monitoring.alerts.panel.disableAlert.errorTitle', {
+          defaultMessage: `Unable to disable alert`,
+        }),
+        text: err.message,
+      });
+    }
+  }
+  async function enableAlert() {
+    try {
+      await Legacy.shims.http.post(`${BASE_ALERT_API_PATH}/alert/${alert.rawAlert.id}/_enable`);
+    } catch (err) {
+      Legacy.shims.toastNotifications.addDanger({
+        title: i18n.translate('xpack.monitoring.alerts.panel.enableAlert.errorTitle', {
+          defaultMessage: `Unable to enable alert`,
+        }),
+        text: err.message,
+      });
+    }
+  }
+  async function muteAlert() {
+    try {
+      await Legacy.shims.http.post(`${BASE_ALERT_API_PATH}/alert/${alert.rawAlert.id}/_mute_all`);
+    } catch (err) {
+      Legacy.shims.toastNotifications.addDanger({
+        title: i18n.translate('xpack.monitoring.alerts.panel.muteAlert.errorTitle', {
+          defaultMessage: `Unable to mute alert`,
+        }),
+        text: err.message,
+      });
+    }
+  }
+  async function unmuteAlert() {
+    try {
+      await Legacy.shims.http.post(`${BASE_ALERT_API_PATH}/alert/${alert.rawAlert.id}/_unmute_all`);
+    } catch (err) {
+      Legacy.shims.toastNotifications.addDanger({
+        title: i18n.translate('xpack.monitoring.alerts.panel.ummuteAlert.errorTitle', {
+          defaultMessage: `Unable to unmute alert`,
+        }),
+        text: err.message,
+      });
+    }
+  }
+
+  const flyoutUi = showFlyout ? (
+    <AlertsContextProvider
+      value={{
+        http: Legacy.shims.http,
+        actionTypeRegistry: Legacy.shims.actionTypeRegistry,
+        alertTypeRegistry: Legacy.shims.alertTypeRegistry,
+        toastNotifications: Legacy.shims.toastNotifications,
+        uiSettings: Legacy.shims.uiSettings,
+        docLinks: Legacy.shims.docLinks,
+        charts: undefined,
+        dataFieldsFormats: undefined,
```

Comment:
You don't really need to do this, since they're optional.

```suggestion
```

Reply:
Oh cool. TIL!